src/CMakeLists.txt

```diff
@@ -190,8 +194,9 @@ if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
     target_link_libraries(cosim INTERFACE "atomic" "stdc++fs")
 endif()
 if(WIN32)
+    target_link_libraries(cosim PUBLIC "Bcrypt")
     if(BUILD_SHARED_LIBS)
-        target_link_libraries(cosim PUBLIC "Bcrypt" "Shlwapi") # Windows API DLLs
+        target_link_libraries(cosim PUBLIC "Shlwapi") # Windows API DLLs
     else()
         set_target_properties(cosim PROPERTIES OUTPUT_NAME "libcosim")
```

Comment:
Shouldn't we have `OUTPUT_NAME "cosim"`?

Reply:
Nope. Windows DLLs come with a companion `.lib` called the "import library" which is what you actually link into your application. The point here is to have different names for the import library and the static library so both can be installed alongside each other. The former is called `cosim.lib` to match the name of `cosim.dll`, while the latter is called `libcosim.lib`, borrowing the *NIX convention. (Boost does it the same way.)

Import libraries are not a thing on Linux, by the way. There, the linker is able to get the required information directly from the `.so` file.